inference-engine/thirdparty/clDNN/src/impls/ocl/non_max_suppression.cpp

```diff
@@ -0,0 +1,183 @@
+            args.inputs.push_back(instance.iou_threshold_mem());
+        }
+
+        if (instance.has_score_threshold() && !instance.node.score_threshold_node().is_constant()) {
+            args.inputs.push_back(instance.score_threshold_mem());
+        }
+
+        if (instance.has_soft_nms_sigma() && !instance.node.soft_nms_sigma_node().is_constant()) {
+            args.inputs.push_back(instance.soft_nms_sigma_mem());
+        }
+
+        args.output = instance.output_memory_ptr();
+        if (instance.has_second_output())
+            args.inputs.push_back(instance.second_output_mem());
+        if (instance.has_third_output())
+            args.inputs.push_back(instance.third_output_mem());
+
+        return args;
+    }
+
+public:
+    static primitive_impl* create(const non_max_suppression_node& arg) {
+        auto params = get_default_params<kernel_selector::non_max_suppression_params>(arg);
+        auto optional_params =
+            get_default_optional_params<kernel_selector::non_max_suppression_optional_params>(arg.get_program());
+
+        const auto& primitive = arg.get_primitive();
+        params.inputs.push_back(convert_data_tensor(arg.input_scores().get_output_layout()));
+
+        if (arg.has_num_select_per_class()) {
+            cldnn::program_node& node = arg.num_select_per_class_node();
+            if (node.is_constant()) {
+                params.num_select_per_class_type = kernel_selector::NmsArgType::Constant;
+                params.num_select_per_class = get_value<int>(node);
+            } else {
+                params.num_select_per_class_type = kernel_selector::NmsArgType::Input;
+                params.inputs.push_back(convert_data_tensor(node.get_output_layout()));
+            }
+        }
+
+        if (arg.has_iou_threshold()) {
+            cldnn::program_node& node = arg.iou_threshold_node();
+            if (node.is_constant()) {
+                params.iou_threshold_type = kernel_selector::NmsArgType::Constant;
+                params.iou_threshold = get_value<float>(node);
+            } else {
+                params.iou_threshold_type = kernel_selector::NmsArgType::Input;
+                params.inputs.push_back(convert_data_tensor(node.get_output_layout()));
+            }
+        }
+
+        if (arg.has_score_threshold()) {
+            cldnn::program_node& node = arg.score_threshold_node();
+            if (node.is_constant()) {
+                params.score_threshold_type = kernel_selector::NmsArgType::Constant;
+                params.score_threshold = get_value<float>(node);
+            } else {
+                params.score_threshold_type = kernel_selector::NmsArgType::Input;
+                params.inputs.push_back(convert_data_tensor(node.get_output_layout()));
+            }
+        }
+
+        if (arg.has_soft_nms_sigma()) {
+            cldnn::program_node& node = arg.soft_nms_sigma_node();
+            if (node.is_constant()) {
```

Comment:
I think you should check here that `node.is_type<data>()` which is not the same as `is_constant()`. Otherwise we may have an exception in `get_value` method